play-frontend-hmrc-play-30/src/test/scala/uk/gov/hmrc/helpers/GeneratorBaseSpec.scala

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.helpers
+
+import org.scalacheck.Arbitrary
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.wordspec.AnyWordSpec
+
+import scala.reflect.ClassTag
+import scala.reflect.runtime.universe._
+
+abstract class GeneratorBaseSpec[T: ClassTag: TypeTag](
+  val ignoredFields: Set[String] = Set.empty,
+  val instanceCount: Int = 100
+)(implicit val arbT: Arbitrary[T])
+    extends AnyWordSpec
+    with Matchers {
+
+  s"Generator for ${typeOf[T]}" should {
+    "generate varying data for all fields" in {
+      val instances       = generateInstances
+      val invariantFields = checkInvariantFieldValues(instances)
+      withClue("Found fields that aren't being randomly generated:") {
+        invariantFields shouldBe empty
+      }
+    }
+  }
+
+  // TODO can't use LazyList because we need to support Scala 2.12
```

Comment:
Maybe just update this to say when we're planning to drop Scala 2.12 supprt (i.e. end May 2024)

Reply:
Yeah, I'd take the `TODO` off here because we can't do it yet - but it's just to warn anyone who sees a deprecation warning and wastes time trying to fix it before we've dropped Scala 2.12